File: latest_ran_output/gpt_5_3_codex/form_002_naive.py
from flask import Flask, request, jsonify
from werkzeug.security import generate_password_hash
import re
import sqlite3
from datetime import datetime

app = Flask(__name__)
DB_PATH = "app.db"


def get_db():
    conn = sqlite3.connect(DB_PATH)
    conn.row_factory = sqlite3.Row
    return conn


def init_db():
    conn = get_db()
    conn.execute(
        """
        CREATE TABLE IF NOT EXISTS users (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            username TEXT NOT NULL UNIQUE,
            email TEXT NOT NULL UNIQUE,
            password_hash TEXT NOT NULL,
            created_at TEXT NOT NULL
        )
        """
    )
    conn.commit()
    conn.close()


init_db()


def is_valid_email(email: str) -> bool:
    pattern = r"^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$"
    return re.match(pattern, email or "") is not None


@app.route("/register", methods=["POST"])
def register():
    data = request.get_json(silent=True) or request.form

    username = (data.get("username") or "").strip()
    email = (data.get("email") or "").strip().lower()
    password = data.get("password") or ""
    confirm_password = data.get("confirm_password") or ""

    if not username or not email or not password:
        return jsonify({"error": "username, email, and password are required"}), 400

    if len(username) < 3 or len(username) > 30:
        return jsonify({"error": "username must be between 3 and 30 characters"}), 400

    if not re.match(r"^[A-Za-z0-9_]+$", username):
        return jsonify({"error": "username may contain only letters, numbers, and underscores"}), 400

    if not is_valid_email(email):
        return jsonify({"error": "invalid email format"}), 400

    if len(password) < 8:
        return jsonify({"error": "password must be at least 8 characters long"}), 400

    if confirm_password and password != confirm_password:
        return jsonify({"error": "passwords do not match"}), 400

    password_hash = generate_password_hash(password)

    conn = get_db()
    try:
        conn.execute(
            """
            INSERT INTO users (username, email, password_hash, created_at)
            VALUES (?, ?, ?, ?)
            """,
            (username, email, password_hash, datetime.utcnow().isoformat() + "Z"),
        )
        conn.commit()
        user_id = conn.execute("SELECT last_insert_rowid()").fetchone()[0]
    except sqlite3.IntegrityError as e:
        msg = str(e).lower()
        if "users.username" in msg or "username" in msg:
            return jsonify({"error": "username already in use"}), 409
        if "users.email" in msg or "email" in msg:
            return jsonify({"error": "email already in use"}), 409
        return jsonify({"error": "account could not be created"}), 409
    finally:
        conn.close()

    return (
        jsonify(
            {
                "message": "account created",
                "user": {"id": user_id, "username": username, "email": email},
            }
        ),
        201,
    )


if __name__ == "__main__":
    app.run(debug=True)

File: latest_ran_output/gpt_5_3_codex/form_002_naive.js
const express = require("express");
const bcrypt = require("bcrypt");
const crypto = require("crypto");

const router = express.Router();

// Example in-memory store (replace with DB in production)
const users = [];

/**
 * POST /register
 * Expects JSON body:
 * {
 *   "name": "Jane Doe",
 *   "email": "jane@example.com",
 *   "password": "StrongPass123!"
 * }
 */
router.post("/register", async (req, res) => {
  try {
    const { name, email, password } = req.body || {};

    // Basic validation
    if (!name || !email || !password) {
      return res.status(400).json({
        success: false,
        message: "Name, email, and password are required.",
      });
    }

    const normalizedEmail = String(email).trim().toLowerCase();

    // Minimal format checks
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    if (!emailRegex.test(normalizedEmail)) {
      return res.status(400).json({
        success: false,
        message: "Invalid email format.",
      });
    }

    if (String(password).length < 8) {
      return res.status(400).json({
        success: false,
        message: "Password must be at least 8 characters long.",
      });
    }

    // Check if user already exists
    const existingUser = users.find((u) => u.email === normalizedEmail);
    if (existingUser) {
      return res.status(409).json({
        success: false,
        message: "An account with this email already exists.",
      });
    }

    // Hash password
    const saltRounds = 12;
    const passwordHash = await bcrypt.hash(password, saltRounds);

    // Create user record
    const user = {
      id: crypto.randomUUID(),
      name: String(name).trim(),
      email: normalizedEmail,
      passwordHash,
      createdAt: new Date().toISOString(),
    };

    users.push(user);

    // Return safe user payload
    return res.status(201).json({
      success: true,
      message: "User account created successfully.",
      user: {
        id: user.id,
        name: user.name,
        email: user.email,
        createdAt: user.createdAt,
      },
    });
  } catch (err) {
    return res.status(500).json({
      success: false,
      message: "Internal server error.",
    });
  }
});

module.exports = router;